modules/objdetect/test/test_barcode.cpp

```diff
@@ -82,6 +85,19 @@ TEST_P(BarcodeDetector_main, interface)
     vector<string> types;
     vector<string> lines;
 
+    // preset parameters based on filename
+    {
+        if(fname.find("_detector_lg.jpg") != string::npos || fname.find("_detector_sm.jpg") != string::npos)
```

Comment:
Can we avoid creating extra test images? 

I suggest the following test scenario:
1. take one of existing images
2. set detector parameters to some predefined values
3. run detector and verify that result is as expected (e.g. it can be empty with some parameter set)

Test can be parametrised with `(params set, expected result)`. Test can be combined with new regression test (check parameters after setting).

Input image can be modified in some way (scaled, rotated, filtered,...) or it can be built from several other images (`vconcat`).

Reply:
> Can we avoid creating extra test images?

Extra test data was designed to detect abnormal changes to the `barcode_detector.cpp` algorithm impl wrt to false positives. It needs to happen.

Reply:
What is different in these images comparing to pre-existing ones? If we really need them, can we reduce their size, for example we could add only one of them (`_sm`) and get crops from it in the test to replace other sizes or maybe somehow extend `_lg` image instead.